cookbook/case_studies/feature_engineering/feast_integration/custom_provider/provider.py

```diff
@@ -0,0 +1,138 @@
+from datetime import datetime
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, Union
+from feast.infra.offline_stores.file_source import FileSource
+from flytekit.core.context_manager import FlyteContext
+
+import pandas
+from feast.entity import Entity
+from feast.feature_table import FeatureTable
+from feast.feature_view import FeatureView
+from feast.infra.local import LocalProvider
+from feast.infra.offline_stores.offline_store import RetrievalJob
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.registry import Registry
+from feast.repo_config import RepoConfig
+from tqdm import tqdm
+
+
+class FlyteCustomProvider(LocalProvider):
+    def __init__(self, config: RepoConfig, repo_path):
+        super().__init__(config)
+
+    def update_infra(
```

Comment:
Do we need all the methods except `get_historical_features()` to be defined (or overridden)? Since they aren't modified, I think we can retain them as is.

Reply:
That's a good callout, will do that. I was thinking about not going with a custom provider and instead have `FlyteFile`-backed offline and online stores. In any case, I'll remove the methods we don't need.